wpilib/wpilib/livewindow.py

```diff
@@ -75,77 +87,75 @@ def setEnabled(enabled):
         after a period of adjusting them in LiveWindow mode.
         """
-        if LiveWindow.liveWindowEnabled != enabled:
+        from .command import Scheduler
+        if cls.liveWindowEnabled != enabled:
+            scheduler = Scheduler.getInstance()
             if enabled:
                 logger.info("Starting live window mode.")
-                if LiveWindow.firstTime:
-                    LiveWindow.initializeLiveWindowComponents()
-                    LiveWindow.firstTime = False
-                Scheduler.getInstance().disable()
-                Scheduler.getInstance().removeAll()
-                bad_components = []
-                for component in LiveWindow.components.keys():
-                    try:
-                        component.startLiveWindowMode()
-                    except Exception as e:
-                        logger.error("Exception running startLiveWindowMode() on {}, removing from component list.".format(component))
-                        logger.exception(e)
-                        bad_components.append(component)
-                for component in bad_components:
-                    del(LiveWindow.components[component])
+                scheduler.disable()
+                scheduler.removeAll()
             else:
                 logger.info("Stopping live window mode.")
-                for component in LiveWindow.components.keys():
-                    component.stopLiveWindowMode()
-                Scheduler.getInstance().enable()
-            LiveWindow.liveWindowEnabled = enabled
-            LiveWindow.statusTable.putBoolean("LW Enabled", enabled)
-
-    @staticmethod
-    def run():
+                for component in cls.components.values():
+                    component.builder.stopLiveWindowMode()
+                scheduler.enable()
+            cls.startLiveWindow = enabled
+            cls.liveWindowEnabled = enabled
+            cls.enabledEntry().setBoolean(enabled)
+
+    @classmethod
+    def run(cls):
         """The run method is called repeatedly to keep the values refreshed
         on the screen in test mode.
+
+        .. deprecated:: 2018.0.0
+            No longer required
         """
-        LiveWindow.updateValues()
+        warnings.warn("run is deprecated. It is no longer required.",
+                      DeprecationWarning, stacklevel=2)
+        cls.updateValues()
 
-    @staticmethod
-    def addSensor(subsystem, name, component):
+    @classmethod
+    def addSensor(cls, subsystem, name, component):
         """Add a Sensor associated with the subsystem and with call it by the
         given name.
 
         :param subsystem: The subsystem this component is part of.
         :param name: The name of this component.
         :param component: A LiveWindowSendable component that represents a
             sensor.
+
+        .. deprecated:: 2018.0.0
+            Use :py:meth:`Sendable.setName(subsystem, name)` instead.
```

Comment:
So I believe by listing the arguments, this doesn't actually hyperlink to the relevant docs.

You can leave off the `:py` at the beginning, the Python domain is the default here.